dsgrid/config/project_config.py

```diff
@@ -468,6 +474,24 @@ def check_for_duplicates(self) -> "RequiredDimensionsModel":
         return self
 
 
+class DatasetBaseDimensionQueryNamesModel(DSGBaseModel):
```

Comment:
This is a key change. We identify these dimension query names when a dataset is submitted to a project and record it in the database. We can later look it up when performing dataset mappings.